File: algorithms/1413.minimum-value-to-get-positive-step-by-step-sum/solution.rs
impl Solution {
    pub fn min_start_value(nums: Vec<i32>) -> i32 {
        let mut min = i32::MAX;

        nums.iter().fold(0, |mut sum, n| {
            sum += n;
            min = min.min(sum);

            sum
        });

        1.max(min * -1 + 1)
    }
}

File: algorithms/1413.minimum-value-to-get-positive-step-by-step-sum/solution.go
func minStartValue(nums []int) int {
	min := math.MaxInt32
	sum := 0

	for _, n := range nums {
		sum += n
		if sum < min {
			min = sum
		}
	}

	if 1 > min*-1+1 {
		return 1
	} else {
		return min*-1 + 1
	}
}
